dbms/src/Storages/DeltaMerge/ConcatSkippableBlockInputStream.cpp

```diff
@@ -247,6 +247,8 @@ Block ConcatVectorIndexBlockInputStream::read()
 {
     load();
     auto block = stream->read();
+    if (!block)
```

Comment:
Interesting, what case will we trigger this?

Reply:
When the block is the last block, it will be empty. And `std::distance(stream->children.begin(), stream->current_stream) = stream->children.size()`, `index` out of range.

Reply:
introduced by https://github.com/pingcap/tiflash/pull/9709